### PR TITLE
docs: (IAC-1005) EKS - Update IAC CDS Postgres CONFIG VARS documentation

### PR DESCRIPTION
**Pull request by dhoucgitter**
# Changes
Update the viya4-iac-aws CONFIG VARS doc to provide an explanation on what a CDS Postgres instance is with links to SAS documentation indicating products that require a cds Postgres instance. Include an example postgres_servers map that shows how to specify both a default and cds-postgres instance in the terraform input file.

### CODE REVIEW

**Review by jarpat — DISMISSED** [review]

**Review by riragh — APPROVED** [review]

File: docs/CONFIG-VARS.md [hunk 1]
```diff
@@ -325,22 +325,23 @@ Each server element, like `foo = {}`, can contain none, some, or all of the para
...
+Here is an example of the `postgres_servers` variable with the `default` server entry overriding only the `administrator_password` parameter and the `cds-postgres` entry overriding all of the parameters:
 
 ```terraform
-database_servers = {
+postgres_servers = {
   default = {
     administrator_password       = "D0ntL00kTh1sWay"
   },
-  another_server = {
+  cds-postgres = {
```
Comment (thpang) [inline docs/CONFIG-VARS.md:337]: I believe the name here must have an underscore for the server instance, but that must be converted to slash when using the value as the server name. Was this taken into account and tested?

**Review by thpang — CHANGES_REQUESTED** [review]

File: docs/CONFIG-VARS.md [hunk 2]
```diff
@@ -325,22 +325,23 @@ Each server element, like `foo = {}`, can contain none, some, or all of the para
 | parameters | additional parameters for PostgreSQL server | list(map(string)) | [] | More details can be found [here](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/Appendix.PostgreSQL.CommonDBATasks.html#Appendix.PostgreSQL.CommonDBATasks.Parameters) |
 | options | additional options for PostgreSQL server | any | [] | |
 
+Multiple SAS offerings require a second PostgreSQL instance referred to as SAS Common Data Store, or CDS PostgreSQL. For more information, see [Common Customizations](https://go.documentation.sas.com/doc/en/itopscdc/default/dplyml0phy0dkr/n08u2yg8tdkb4jn18u8zsi6yfv3d.htm#p0wkxxi9s38zbzn19ukjjaxsc0kl). A list of SAS offerings that require CDS PostgreSQL is provided in [SAS Common Data Store Requirements](https://go.documentation.sas.com/doc/en/itopscdc/default/itopssr/p06lfgkwib3zxbn1t6nyihexp12n.htm#n03wzanutmc6gon1val5fykas9aa). To create and configure an external CDS PostgreSQL instance in addition to the external platform PostgreSQL instance named `default`, specify `cds-postgres` as a second PostgreSQL instance, as shown in the example below.
```
Comment (supear) [inline docs/CONFIG-VARS.md:328]: [Common Customizations] https://documentation.sas.com/?cdcId=itopscdc&cdcVersion=default&docsetId=dplyml0phy0dkr&docsetTarget=n08u2yg8tdkb4jn18u8zsi6yfv3d.htm#p0wkxxi9s38zbzn19ukjjaxsc0kl

File: docs/CONFIG-VARS.md (hunk 1 above)
Comment (dhoucgitter) [reply docs/CONFIG-VARS.md:337]: I haven't encountered any issues using dashes with the postgres server instance names in either terraform or ansible-vars.yaml on AWS or Azure. I verified with Maria that she has not encountered issue like that either and included the tfvars and ansible-vars.yaml she is using on Azure.

(https://rndjira.sas.com/browse/IAC-878 involved creating and using both external postgres instances to validate expected behavior with sas-risk-cirrus products that had just enable support for external CDS)
The terraform-input.tfvars content pulled from local disk that was used to successfully create both external postgres instances in AWS looks like this:

```yaml
postgres_servers = {
  default = {
    ssl_enforcement_enabled = true
    },
  cds-postgres = {
    ssl_enforcement_enabled = true
    }
}
```
and the AWS ansible-vars.yaml values used with those external pg instances are as follows:

```yaml
V4_CFG_POSTGRES_SERVERS:
  default:
    internal: false
  cds-postgres:
    internal: false
```

Maria C shared these examples of terraform-input.tfvars and ansible-vars.yaml values that she has been using successfully on Azure.

```yaml
postgres_servers = {
  default = {
    sku_name                     = "MO_Standard_E8ds_v4"
    storage_mb                   = 65536
    administrator_login          = "pgadmin"
    administrator_password       = "Go4thsas"
    server_version               = "14"
    postgresql_configurations    = [
       {
         name  = "azure.extensions"
         value = "PLPGSQL,LTREE"
       }
      ]
  },
  cds-postgres = {
    sku_name                     = "MO_Standard_E8ds_v4"
    storage_mb                   = 65536
    administrator_login          = "pgadmin"
    administrator_password       = "Go4thsas"
    server_version               = "14"
    postgresql_configurations    = [
       {
         name  = "azure.extensions"
         value = "PLPGSQL,LTREE"
       }
      ]
  }
}
```

and the ansible-vars.yaml pg values used with those external pd instances

```yaml
V4_CFG_POSTGRES_SERVERS:
  default:
    internal: false
    admin: pgadmin
    password: Go4thsas
    fqdn: maria-iacfips-default-flexpsql.postgres.database.azure.com
    server_port: 5432
    ssl_enforcement_enabled: true
    database: SharedServices
  cds-postgres:
    internal: false
    admin: pgadmin
    password: Go4thsas
    fqdn: maria-iacfips-cds-postgres-flexpsql.postgres.database.azure.com
    server_port: 5432
    ssl_enforcement_enabled: true
```

Jay mentioned this issue from JK https://rndjira.sas.com/browse/IAC-1022 on GCP related to creating 2 or more external PG instances but it is unrelated to dashes being in the name.

File: docs/CONFIG-VARS.md (hunk 2 above)
Comment (dhoucgitter) [reply docs/CONFIG-VARS.md]: updated links from Susan applied.

File: docs/CONFIG-VARS.md (hunk 1 above)
Comment (thpang) [reply docs/CONFIG-VARS.md:337]: From the postgres docs: https://www.postgresql.org/docs/7.0/syntax525.htm#:~:text=Names%20in%20SQL%20must%20begin,but%20they%20will%20be%20truncated. You cannot use a '-' in the name but can use an '_'. For the DNS entry for the server you must use a '-' and cannot use an '_'. Not sure the cloud providers have altered this one, but if it's been tested just be aware of the consequences.

**Review by thpang — APPROVED** [review]
Comment in PR about the sample name. Hesitant to approve but doing so as this item has been brought up before.

File: docs/CONFIG-VARS.md (hunk 1 above)
Comment (dhoucgitter) [reply docs/CONFIG-VARS.md:337]: Thanks for posting that link, here's the link to that section for latest supported version, Postgres 15, https://www.postgresql.org/docs/15/sql-syntax-lexical.html#SQL-SYNTAX-IDENTIFIERS.

**Review by riragh — APPROVED** [review]

**Pull request merged**